### PR TITLE
LPD-17027 Unsafe suppliers on optional fields (Objects)

### PR DESCRIPTION
**Pull request by albertojml**
**Purpose of this PR**
The main objective of this PR is improve the Objects APIs performance by using a new "lazyProperties" mechanism. This String-UnsafeSupplier map is accessed while recovering the data fields required without retrieving the optional fields which results in less queries and better performance for the APIs for the different Object entries.

**What's changed?:**
- Now, through Rest-Builder, every Json Map property gets an analog lazy map which will be accessed while retrieving the properties. This includes its respective getters and setters.
- The ObjectsDTOConverter has been adapted to build the Object Entries DTOs so that this lazy mechanism gets implemented.
- A new serializer for Unsafe Suppliers has been created.
- The objects nested fields infrastructure has been changed to take into account the new lazy mechanism.
- Batch Engine is adapted for this new mechanism.

**What's new?:**
- A new serializer for Unsafe Suppliers has been created.
- New tests for checking that the unsafe suppliers are not invoked when they're not needed.

### CODE REVIEW

**Comment by albertojml** [comment]
ci:test:sf

**Comment by albertojml** [comment]
ci:test:sf

**Comment by albertojml** [comment]
ci:stop

**Comment by albertojml** [comment]
ci:test:sf

**Comment by albertojml** [comment]
ci:test:relevant

**Comment by albertojml** [comment]
ci:test:sf

**Comment by albertojml** [comment]
ci:test:relevant

**Review by 4lejandrito — APPROVED** [review]

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
Could you @matijapetanjek please check the Batch Engine part implemented in Objects?

File: modules/apps/batch-engine/batch-engine-service/src/test/java/com/liferay/batch/engine/internal/writer/ColumnValuesExtractorTest.java [hunk 1]
```diff
@@ -218,14 +218,17 @@ private class ColumnDescriptorProviderImpl
 		@Override
 		public ColumnDescriptor[] getColumnDescriptors(
 			long companyId, String fieldName, int index,
-			ObjectValuePair<Field, Method> propertiesObjectValuePair,
+			Map<String, ObjectValuePair<Field, Method>> objectValuePairs,
```
Comment (ccorreagg) [inline modules/apps/batch-engine/batch-engine-service/src/test/java/com/liferay/batch/engine/internal/writer/ColumnValuesExtractorTest.java:221]: Now we have a map but we are just accesing to one of the values. Is this change necessary?

**Comment by ccorreagg** [comment]
@albertojml I have included some commits and also a question, because I don't understand that change, maybe it is something that we had in previous commits but not needed anymore. Please have a look at my commits and see if you agree with those changes

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by albertojml** [comment]
ci:test:sf

**Comment by albertojml** [comment]
ci:test:relevant

**Comment by albertojml** [comment]
ci:test:relevant

**Comment by albertojml** [comment]
Reruning the Relevant test suite because of [this test failure](https://testray.liferay.com/home/-/testray/case_results/2561467919). It passed on local.

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
This change should not exist:

image

I think it is noisy @4lejandrito

**Comment by albertojml** [comment]
ci:test:sf

**Comment by albertojml** [comment]
ci:test:relevant

**Comment by albertojml** [comment]
ci:test:relevant

**Comment by albertojml** [comment]
ci:test:sf

File: modules/apps/batch-engine/batch-engine-service/src/test/java/com/liferay/batch/engine/internal/writer/ColumnValuesExtractorTest.java (hunk 1 above)
Comment (matijapetanjek) [reply modules/apps/batch-engine/batch-engine-service/src/test/java/com/liferay/batch/engine/internal/writer/ColumnValuesExtractorTest.java:221]: Admittedly I don't understand the bigger picture of these changes but I fail to see how refactoring ObjectValuePair<Field, Method> propertiesObjectValuePair -> Map<String, ObjectValuePair<Field, Method>> objectValuePairs will prevent "leaking of object properties"

Comment (4lejandrito) [reply modules/apps/batch-engine/batch-engine-service/src/test/java/com/liferay/batch/engine/internal/writer/ColumnValuesExtractorTest.java:221]: Only objects has the concept of properties. If we ever need to implement this interface for other entities we will not have anything to pass there. The concept of `properties` is unique to objects and it should not leak in the interface. By passing all the value pairs we make it future proof.

Comment (4lejandrito) [reply modules/apps/batch-engine/batch-engine-service/src/test/java/com/liferay/batch/engine/internal/writer/ColumnValuesExtractorTest.java:221]: This was made evident when we renamed `properties` to `lazyProperties`. It made no sense to have that outside of the objects implementation.

**Comment by albertojml** [comment]
ci:test:sf

**Comment by albertojml** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Review by ccorreagg — APPROVED** [review]

**Comment by albertojml** [comment]
> Please @albertojml have a look at my latest commits and let me know what you think. Apart from that, there was a commit from Alex with a change in the ObjectEntry.java class that I have included into the rest builder commit, otherwise we would register a commit in an autogenerated class

@ccorreagg Looks good to me! If no weird errors are thrown in the relevant suite the PR can be sent. Thanks!

**Comment by ccorreagg** [comment]
@albertojml GREAT JOB

**Comment by albertojml** [comment]
ci:test:sf

**Comment by albertojml** [comment]
ci:test:relevant

**Comment by albertojml** [comment]
ci:test:sf

**Comment by albertojml** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:report:2562457739

**Comment by albertojml** [comment]
@ccorreagg I will take a look. Thanks for pointing it out!

**Comment by albertojml** [comment]
ci:test:relevant

**Comment by albertojml** [comment]
ci:forward
